dist/index.d.ts

```diff
@@ -3880,68 +3883,92 @@ interface RoomPosition {
      * @param opts An object containing pathfinding options (see Room.findPath), or one of the following: filter, algorithm
      * @returns An instance of a RoomObject.
      */
-    findClosestByPath<K extends FindConstant, S extends FindTypes[K]>(
+    findClosestByPath<K extends FindConstant, T extends FindTypes[K], S extends T>(
         type: K,
-        opts?: FindPathOpts & Partial<FilterOptions<K, S>> & { algorithm?: FindClosestByPathAlgorithm },
+        opts?: FindPathOpts & Partial<PredicateFilterOptions<T, S>> & { algorithm?: FindClosestByPathAlgorithm },
     ): S | null;
-    findClosestByPath<S extends AnyStructure>(
+    findClosestByPath<K extends FindConstant, T extends FindTypes[K] = FindTypes[K]>(
+        type: K,
+        opts?: FindPathOpts & Partial<FilterOptions<FindTypes[K]>> & { algorithm?: FindClosestByPathAlgorithm },
+    ): T | null;
+    findClosestByPath<T extends AnyStructure>(
```

Comment:
I would rather delete this declaration than keep this. But I have seen many player who use find<StructureFoobar>, so keep this for compatibility

Reply:
Let's keep it as is now.